src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java

```diff
@@ -40,83 +42,94 @@
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 /**
- * X509 grant type for Identity Server
+ * ssh-ed25519-cert-v01 grant type for Identity Server
  */
-public class X509GrantHandler extends AbstractAuthorizationGrantHandler  {
+public class X509GrantHandler extends AbstractAuthorizationGrantHandler {
 
-    public static final String SSH_CERT_TYPE_ED25519 = "ssh-ed25519-cert-v01@openssh.com";
+    private static final String SSH_CERT_TYPE_ED25519 = "ssh-ed25519-cert-v01@openssh.com";
     private static Log log = LogFactory.getLog(X509GrantHandler.class);
-    public static final String X509_GRANT_PARAM = "x509";
+    private static final String X509_GRANT_PARAM = "x509";
 
     @Override
-    public boolean validateGrant(OAuthTokenReqMessageContext oAuthTokenReqMessageContext)  throws
+    public boolean validateGrant(OAuthTokenReqMessageContext oAuthTokenReqMessageContext) throws
             IdentityOAuth2Exception {
 
-        boolean authStatus = false;
-        // extract request parameters
-        RequestParameter[] parameters = oAuthTokenReqMessageContext.getOauth2AccessTokenReqDTO().getRequestParameters();
-        String certParam = null;
-        String subjectDN = null;
-        // find out subjectDN
-        for(RequestParameter parameter : parameters){
-            if(X509_GRANT_PARAM.equals(parameter.getKey())){
-                if(parameter.getValue() != null && parameter.getValue().length > 0){
-                    certParam = parameter.getValue()[0];
-                }
-            } else if("subjectDN".equalsIgnoreCase(parameter.getKey())) {
-                if(parameter.getValue() != null && parameter.getValue().length > 0){
-                    subjectDN = parameter.getValue()[0];
-                }
+        String certificateParameterValue = null;
+        String username = null;
+
+        // Extract request parameters.
+        RequestParameter[] requestParameters = oAuthTokenReqMessageContext.getOauth2AccessTokenReqDTO()
+                .getRequestParameters();
+
+        // Extract 'x509' param from the request.
+        for (RequestParameter parameter : requestParameters) {
+            if (X509_GRANT_PARAM.equals(parameter.getKey()) && !ArrayUtils.isEmpty(parameter.getValue())) {
+                certificateParameterValue = parameter.getValue()[0];
+                break;
             }
         }
 
-        String username = null;
-        if(subjectDN != null) {
-            username = subjectDN;
-            log.debug("Username is retrieved from Subject DN : " + username);
-        } else {
-            if (certParam != null) {
-                username = getUserFromSSHCert(certParam);
-            }
+        // Extract 'Principal' from the presented certificate.
+        if (certificateParameterValue != null) {
```

Comment:
Let's throw an error if `certificateParameterValue` is `null`

Reply:
Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/3/commits/d01c3a2846a13850ced1a2e93b47bb46687eb8e1